### DISCUSSION TITLE
Rendering using GPU, how the right approach?

### DISCUSSION THREAD

**Discussion by ArturLourenco** [tylermorganwall/rayshader#311] [Q&A]
Hi,

I have been trying to render my rayshader plots using my GPU (NVIDIA RTX A1000). I'm using windows OS right now and tried a lot of configurations to force R, Rstudio, RGL... to render using GPU. In some cases when I use plot3d() function I see that the rendering process is under GPU, in this code por example:

```
elmat%>%
  height_shade(texture = topo.colors(256)) %>%
  add_shadow(hillshade, 0.5) %>%
  plot_3d(elmat, zscale = .3, fov = 0, theta = 135, phi = 45, windowsize = c(1000, 800),
          soil = TRUE) 
``` 

If I add   add_water(detect_water(elmat), color = "desert") %>% in this code the plot3D won't run under my GPU anymore. render_highquality() function never runs using the GPU. I would like to know by the community what are the approach's used to render plots using rayshaderverse plus NVIDIA GPUs.

**Comment by Bhavya031** [comment]
i have the same issue without gpu 3d render takes too long.
